src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/LanguageServerFeatureOptions.cs

```diff
@@ -34,4 +34,13 @@ internal abstract class LanguageServerFeatureOptions
     /// ensure a unique file path per project.
     /// </summary>
     public abstract bool IncludeProjectKeyInGeneratedFilePath { get; }
+
+    /// <summary>
+    /// Whether to monitor the entire workspace folder for any project.razor.bin files
+    /// </summary>
+    /// <remarks>
+    /// When this is off, the language server won't have any project knowledge unless the
+    /// razor/monitorProjectConfigurationFilePath notification is sent.
+    /// </remarks>
+    public abstract bool MonitorWorkspaceFolderForConfigurationFiles { get; }
```

Comment:
might there be issues with disco colors when this flag is turned off? is this used in vs code too?

Reply:
> Tested and confirmed when loading a solution with project.razor.vs.bin files in both obj\Debug\net7.0 and obj\Debug\net8.0, we correctly only load the net8.0 project. Verified by checking debug logs for the string "net7" and not finding it :)

This comment seems related. Thanks.

Reply:
VS Code does not set this flag. PRs https://github.com/dotnet/roslyn/pull/70913 and https://github.com/dotnet/vscode-csharp/pull/6688 are open to address that.